### PR TITLE
Update delete all to include trash

### PR DESCRIPTION
**Pull request by rrennick**
### Changes proposed in this Pull Request:

This PR updates the `deleteAllObjects` function to also delete trashed objects. It also updates the product import test to use the api to delete the imported products.

### How to test the changes in this Pull Request:

1. Run E2E tests in `trunk`
2. Note that there are products in the trash
3. Switch to this branch
4. Run `npm run build:packages`
5. Run the merchant tests `npx wc-e2e tests/e2e/specs/wp-admin/merchant.js`
6. There will be published products left from the tests but the trashed products should be deleted 

### Changelog entry

> N/A

### CODE REVIEW

**Comment by lsinger** [comment]
@rrennick I tried to review this but after running the merchant tests I still have 37 products in the Trash. Just to make sure we're on the same page: 

1. run full e2e test suite in trunk
2. note products in trash
3. switch to this PR's branch
4. run merchant tests `npx wc-e2e test:e2e tests/e2e/specs/wp-admin/merchant.js`

I also have two tests fail in step 4:

```
  Add New Variable Product Page
    ✓ can create product with variations (1466ms)
    ✓ can create set variable product attributes (2876ms)
    ✓ can create variable product variations (1874ms)
    ✕ can create variation attributes (30004ms)
  WooCommerce General Settings
    ✕ can update settings (7767ms)
```

File: tests/e2e/utils/src/flows/with-rest-api.js [hunk 1]
```diff
@@ -16,20 +16,24 @@ const userEndpoint = '/wp/v2/users';
...
 	const minimum = defaultObjectId == null ? 0 : 1;
+	const statuses = ['publish','trash'];
 
-	objects = await repository.list();
-	while ( objects.length > minimum ) {
-		for (let o = 0; o < objects.length; o++ ) {
-			// Skip default data store object
-			if ( objects[ o ].id == defaultObjectId ) {
-				continue;
+	for ( let s = 0; s < statuses.length; s++ ) {
```
Comment (lsinger) [inline tests/e2e/utils/src/flows/with-rest-api.js:21]: Using `statuses.each` here might  back it a bit more readable.

Comment (rrennick) [reply tests/e2e/utils/src/flows/with-rest-api.js:21]: > Using statuses.each here might back it a bit more readable.

I had that and was getting the warning that the promises would be ignored and to use the basic for loop instead.

Comment (lsinger) [reply tests/e2e/utils/src/flows/with-rest-api.js:21]: Ah! Right. Would need an async `each` then but yes, that doesn't sound enticing.

**Comment by rrennick** [comment]
> I tried to review this but after running the merchant tests I still have 37 products in the Trash.

@lsinger Sorry about that. I edited the description and added a couple steps to the testing instructions.

**Review by lsinger — APPROVED** [review]
When I ran the merchant tests all by themselves, four of them failed: 

```
  WooCommerce Orders > Add new order
    ✕ can create new order (5486ms)
    ✕ can create new complex order with multiple product types & tax classes (1325ms)
```

```
  Add New Variable Product Page
    ✓ can create product with variations (1495ms)
    ✓ can create set variable product attributes (2886ms)
    ✓ can create variable product variations (1896ms)
    ✕ can create variation attributes (30006ms)
  WooCommerce General Settings
    ✕ can update settings (7715ms)
```

There were no trashed products, only 9 published and 1 draft. 👍 

When running all e2e tests on this branch they all succeed. 👍

**Pull request merged**